### PR TITLE
Add nearest_valid_timecode_rate function

### PR DESCRIPTION
**Pull request by splidje**
Added useful function nearest_valid_timecode_rate in lieu of access to the list of valid timecode rates in Python.

Example situation:

You've for some reason lost a significant figure on a frame rate: 23.97602397602397

`is_valid_timecode_rate(23.97602397602397)` returns `False`

`nearest_valid_timecode_rate(23.97602397602397)` returns `23.976023976023978`

### CODE REVIEW

**Review by meshula — APPROVED** [review]
looks like a good addition.

**Comment by jminor** [comment]
This looks useful and clear. Could you add a test and a doc string for the new function?

File: tests/test_opentime.py [hunk 1]
```diff
@@ -685,6 +685,19 @@ def test_passing_ndf_tc_at_df_rate(self):
         t2 = otio.opentime.from_timecode(NDF_TC, 29.97)
         self.assertEqual(t2.value, frames)
 
+    def test_nearest_valid_timecode_rate(self):
+        invalid_rate = 23.97602397602397
+        nearest_valid_rate = 23.976023976023978
+
+        self.assertFalse(otio.opentime.RationalTime.is_valid_timecode_rate(invalid_rate))
```
Comment (JeanChristopheMorinPerso) [inline tests/test_opentime.py:692]: Builds show a linting error here: https://github.com/PixarAnimationStudios/OpenTimelineIO/runs/4704135694?check_suite_focus=true#step:7:13

**Comment by jminor** [comment]
Could you add test cases for the common "abbreviated" NTSC rates. Specifically, 29.97, 23.98, 23.976 to ensure that those snap to the desired NTSC rate. Those truncated rates often appear in text fields, labels, etc. so they might find their way into this function.

**Comment by splidje** [comment]
29.97, 23.98 and 23.976 are already in the valid timecode rates list.

Did you mean a test which checks e.g. 29.971 snaps to 29.97?

**Comment by reinecke** [comment]
Ahh, right. I think this is a place where we might want to apply the [Robustness Principal](https://en.wikipedia.org/wiki/Robustness_principle) - we want to be permissive in what we accept (slightly rounded SMPTE rates are okay), but make our functions yield the technically correct SMPTE rates.

**Comment by splidje** [comment]
Ok, so I guess I could make a separate array of the standards, and an array of acceptable roundings, then the `nearest_` function uses the former array?

**Comment by splidje** [comment]
ok, sweet. I'll do that.

**Comment by splidje** [comment]
Just a thought. Would it make sense to have a type like `RationalRate` to accurately represent a drop frame rate like 3000 / 1001 with the numerator and denominator?

**Comment by reinecke** [comment]
Moving RationalTime to be a true rational number is something we've discussed, #190 has some more detail.
@ssteinbach and @meshula are spearheading some work to rigorously define OTIO's underlying mathematical framework. I suspect once that solidifies a bit we'll focus on discussing "RationalTime 2" - we'd certainly love to hear your perspectives!

**Comment by meshula** [comment]
Yes, we have defined some rigorous math, and we have a prototype implementation that correctly (according to the unit tests at least) implements an integer rational number class that guarantees a certain precision, and tells you when you will lose precision by combining rates.

We've also got documentation under way to nail down nomenclature, so that we can all mean the same thing when we use terms of art, such as pull down, and so on.

All of this will show up at a tsc and the github repo during the course of this year, I expect.

**Pull request merged**